src/clients/KubectlClient.ts

```diff
@@ -100,62 +100,48 @@ export class KubectlClient implements IClient {
+            let containerList: string[] = [];
+            const k8sClient = await this._accountContextManager.getK8sClient();
+            const response = await k8sClient.listNamespacedPod(namespace);
+            const pods = response.body.items;
+            podNameList.forEach(podName => {
+                const arr = pods
+                .find(pod => pod.metadata.name === podName)
+                ?.spec?.containers?.map(container => container.name);
+                if (arr && arr.length > 0) {
+                    containerList = containerList.concat(arr);
+                }
+            });
+
+            return [...new Set(containerList)]; // remove duplicates
         } catch (error) {
             this._logger.error(TelemetryEvent.KubectlClient_GetContainerListError, error);
             return null;
         }
 
     }
 
-    public async getPodName(serviceName: string): Promise<string> {
+    public async getPodName(serviceName: string, namespace: string): Promise<string[]> {
         try {
-            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
-            // find the pod ipaddressList for the selected service
-            const args: string[] = [`get`, `ep`, serviceName, `-o`, `jsonpath='{.subsets[*].addresses[*].ip}'`];
-            let output = await this.runKubectlCommandAsync(args, kubeconfigPath);
-            // if there replicas and multiple pods, split it by apostrope, space example value: ''10.2.45.6 10.5.6.809''
-            // handles single value as well example value: ''10.56.78.90''
-            let ipaddressList: string[];
-            if (output && output.length > 0) {
-                ipaddressList = output.replace(/'/g, '').replace(/[\s,]+/g, ',').split(',');
-            }
-
-            if (ipaddressList?.length > 0) {
-                // find the podname by looping the ip address
-                for (const ip in ipaddressList) {
-                    const args2: string[] = [`get`, `pods`];
-                    const fieldSelector = '--field-selector=status.podIP='.concat(ipaddressList[ip]);
-                    args2.push(fieldSelector);
-                    args2.push("-o=name");
-                    const finalOutput = await this.runKubectlCommandAsync(args2, kubeconfigPath);
-                    // split the output by forward slash and new line char at the end example value: 'pod/stats-api-ff7d66c5b-4nc9x\n' 
-                    // output is stats-api-ff7d66c5b-4nc9x
-                    if (finalOutput?.length > 0) {
-                        return finalOutput.indexOf("/") != -1 ? finalOutput.split("/")[1].split("\n")[0] : finalOutput.split("\n")[0];
-                    }
-                }
-            }
-            return null;
+            const k8sClient = await this._accountContextManager.getK8sClient();
+            const resp = await k8sClient.listNamespacedEndpoints(namespace);
```

Comment:
We can use `readNamespacedEndpoints` here to get a single `Endpoints` resource object, rather than listing all the ones in the namespace. See #5